### DISCUSSION TITLE
Add rotating queue mode

### DISCUSSION THREAD

**Discussion by respektive** [ppy/osu#15900] [Ideas] [answered]
The way the current round robin queue works is kinda weird as it can happen that there are 2 players playing for hours and a new person joins and will basically get higher priority, meaning the first 2 players won't be able choose a map again in a long time.

I think a queue mode where it literally just rotates through the users and takes their oldest, not played, queued up map, so that the same user can never pick 2 maps in a row (unless they're the only person in the room ofc.) would be good. This would replace the "host rotate" mode that people do manually in stable multi right now and it's probably what most people are currently expecting round robin to do.

**Comment by peppy** [comment, marked as answer]
That should be how round robin works, as already discussed. It should not be a new mode.

**Reply by respektive** [reply]
alright, yeah i wasn't sure if that's exactly how it should work.

**Reply by smoogipoo** [reply]
I've unmarked this as an answer because the issue is still valid. It doesn't properly handle the described situation though that's how it's _supposed_ to work.

Edit: Nevermind, issue made for this.

**Comment by Theighlin** [comment]
Writing here as this is a discussion to improve the already present queue mode at this point i take it.

In addidion to the proposal i think a players' enqued items should be removed if said player leaves the lobby, we shouldn't play maps of someone that is no longer there.

I know the limit for enqued beatmaps is 3 currently, this works for now as lazer's lobbies are smaller than stable's. Maybe we could lower it for bigger lobbies? In stable, rooms that are full of players already take up to an hour to have a full cycle (each player has selected and played a beatmap of his choice) between playing and waiting for players to ready up. I think in such cases enqueing more than one map at a time might not be desirable?

**Reply by respektive** [reply]
Agree on the maps being removed when a player leaves. About the limit, i don't think that actually matters that much, but there should be a way to edit/remove already enqueued items.

**Reply by peppy** [reply]
The limit will be configurable in the future.

As for not playing beatmaps from users that left: the host will have the ability to skip or remove beatmaps in the very near future. That should cover this problem.

**Reply by Theighlin** [reply]
I'm fine with it i guess, if the map was good and players want to play it they should be able to. 
I think adding an indicator for it would be a good solution? Tint the user icon in the beatmap panel red or something if he leaves (and undo it if he rejoins), would this be okay?
